File: framework/src/main/java/com/software/dm/swallow/stormy/algoac/impl/ByteState.java
package com.software.dm.swallow.stormy.algoac.impl;

import com.software.dm.swallow.stormy.algoac.inter.AbstractState;

/**
 * @author DearM
 * @version v1.0.0.1
 * @Description
 * @date 2016
 */
public class ByteState extends AbstractState<Byte> {

    public ByteState() {
        super();
    }

    public ByteState(int depth) {
        super(depth);
    }


    public AbstractState<Byte> addState(Byte c) {
        AbstractState<Byte> abstractState = this.nextStateIgnoreRootState(c);
        if (abstractState == null) {
            abstractState = new ByteState(this.depth + 1);
            this.success.put(c, abstractState);
        }
        return abstractState;
    }


}

File: framework/src/main/scala/com/software/dm/swallow/stormy/scala/algoac/impl/ByteState.scala
package com.software.dm.swallow.stormy.scala.algoac.impl

import com.software.dm.swallow.stormy.scala.algoac.inter.AbstractState


/**
  *
  * Description
  * @author DearM
  * date 2016
  * @version v1.0.0.1
  *
  */
class ByteState(  depth: Int) extends AbstractState[Byte]( depth: Int)  {


  override def addState(c: Byte): AbstractState[Byte] = {
    var abstractState = this.nextStateIgnoreRootState(c)
    if (abstractState == null) {
      abstractState = new ByteState(this.depth + 1)
      this.success.put(c, abstractState)
    }
    abstractState
  }
}